apps/namadillo/src/lib/query.ts

```diff
@@ -91,22 +91,30 @@ export const buildTxArray = async <T>(
 
   if (!publicKey) {
     const revealPkTx = await tx.buildRevealPk(wrapperTxProps, account.address);
-    txArray.push({ type: revealPublicKeyType, encodedTx: revealPkTx });
+    txArray.push({ type: revealPublicKeyType, tx: revealPkTx.tx });
   }
 
   const encodedTxs = await Promise.all(
     queryProps.map((props) => txFn.apply(tx, [wrapperTxProps, props]))
   );
 
-  const typedEncodedTxs = encodedTxs.map((encodedTx, index) => ({
-    encodedTx,
+  const initialTx = encodedTxs[0].tx as BuiltTx;
```

Comment:
I think this will compile without casting with `as BuiltTx`, and also in the call to `buildBatch` below.

Reply:
I will take a look! I'd really like to get rid of this `BuiltTx` entirely and clean up some of this Tx code. I feel it is unnecessarily complicated

Reply:
ah, yeah, I was messing with having a separate type for batches but removed it, so that was something I forgot to change back :)